File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 1
        k = 0

        for i in range(len(nums)):
            if i > 0 and nums[i] == nums[i-1]:
                count += 1
            else:
                count = 1
            
            if count <= 2:
                nums[k] = nums[i]
                k += 1
        
        return k

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int k = 1;
        int count = 1;

        for(int i = 1; i < size(nums); i++){
            if(nums[i-1] == nums[i]){
                count++;
            }
            else{
                count = 1;
            }

            if(count <= 2){
                nums[k] = nums[i];
                k++;
            }
        }

        return k;
        
    }
};